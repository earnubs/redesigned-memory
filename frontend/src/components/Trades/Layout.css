.eb-Layout-Box {
  background-color: #fff;
  border-radius: 5px;
  box-shadow: 0 2px 4px 0px rgba(0,0,0,0.10);
  padding: 2rem;
  margin: 2rem auto;
}

.eb-Layout-screen {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(243,249,252,0.8);
  z-index: 10000;
}
