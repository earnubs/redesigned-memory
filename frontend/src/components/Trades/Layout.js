import React from 'react';
import PropTypes from 'prop-types';
import { Route } from 'react-router-dom';

import './Layout.css';

export function Box({ children, ...rest }) {
  return (
    <div className="eb-Layout-Box" { ...rest }>
      { children }
    </div>
  );
}

Box.propTypes = {
  children: PropTypes.node.isRequired,
};

function Layout({ tradesList, newTrade }) {
  return (
    <div className="eb-Layout">
      <Box style={{width: 860 }}>
        { tradesList}
      </Box>
      <Route exact path="/new" render={props => (
        <div className="eb-Layout-screen">
          <Box style={{
            width: 400,
            position: 'absolute',
            top: '50%',
            left: '50%',
            transform: 'translate(-50%, -50%)',
          }}>
            { React.cloneElement(newTrade, props) }
          </Box>
        </div>
      )} />
    </div>
  );
}

Layout.defaultProps = {};

Layout.propTypes = {
  tradesList: PropTypes.element.isRequired,
  newTrade: PropTypes.element.isRequired,
};

export default Layout;
