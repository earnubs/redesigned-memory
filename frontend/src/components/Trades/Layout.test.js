import React from 'react';
import { shallow } from 'enzyme';

import Layout, { Box } from './Layout';

describe('Layout components', () => {
  it('should render <Box/> as expected', () => {
    const wrapper = shallow(<Box>Test</Box>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Layout /> as expected', () => {
    const wrapper = shallow(
      <Layout
        tradesList={<div>List</div>}
        newTrade={<div>New</div>}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
