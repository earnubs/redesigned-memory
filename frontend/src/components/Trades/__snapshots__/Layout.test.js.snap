// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout components should render <Box/> as expected 1`] = `
<div
  className="eb-Layout-Box"
>
  Test
</div>
`;

exports[`Layout components should render <Layout /> as expected 1`] = `
<div
  className="eb-Layout"
>
  <Box
    style={
      Object {
        "width": 860,
      }
    }
  >
    <div>
      List
    </div>
  </Box>
  <Route
    exact={true}
    path="/new"
    render={[Function]}
  />
</div>
`;
